### PR TITLE
VTTablet Modes

### PR DESCRIPTION
**Pull request by sougou**
Long overdue documentation on how to launch vttablet in various modes.

### CODE REVIEW

File: doc/VTTabletModes.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+It will also load other information from the my.cnf, like the location of data files, etc. When it receives a request to take a backup, it will shut down mysql, copy the mysql data files to the backup store, and restart mysql.
+
+The my.cnf file can be specified the following ways:
+
+* Implicit: If mysql was initialized by the `mysqlctl` tool, then vttablet can find it based on just the `-tablet-path`. The default location for this file is `$VTDATAROOT/vt_<tablet-path>/my.cnf`.
+* `-mycnf-file`: This option can be used if the file is not present in the default location.
+* `-my_cnf_server_id` and other flags: You can specify all values of the my.cnf file from the command line, and vttablet will behave as it read this information from a physical file.
```
Comment (dkhenry) [inline doc/VTTabletModes.md]: You should specify the three files that form the base my.cnf (master.cnf,default.cnf,replica.cnf). Also aside from `mycnf-file` you can pass in the  EXTRA_MY_CNF env flag to add in additional files. And at some place, probbably not here we should list out how vitess uses the go template language to construct the final my.cnf

Comment (sougou) [reply doc/VTTabletModes.md]: These are for mysqlctl. VTTablet doesn't care about them. I'll remember to mention them while documenting mysqlctl.

File: doc/VTTabletModes.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+### -restore_from_backup
+
+The default value for this flag is false. If set to true, and the my.cnf file was successfully loaded, then vttablet can perform automatic restores as follows:
+
+If started against a mysql instance that has no data files, it will search the list of backups for the latest one, and initiate a restore. After this, it will point the mysql to the current master and wait for replication to catch up. Once replication is caught up to the specified tolerance limit, it will advertise itself as serving. This will cause the vtgates to add it to the list of healthy tablets to serve queries from.
+
+If this flag is true, but my.cnf was not loaded, then vttablet will fatally exit with an error message.
+
+You can additionally control the level of concurrency for a restore with the `-restore_concurrency` flag. This is typically useful in cloud environments to prevent the restore process from becoming a 'noisy' neighbor by consuming all available disk IOPS.
+
```
[the remark this answers is not in the document]
Comment (sougou) [reply doc/VTTabletModes.md]: Same as above: init_db is for mysqlctl.

File: doc/VTTabletModes.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+-backup_storage_implementation file
+-file_backup_storage_root $BACKUP_MOUNT
+-restore_from_backup
+-vtctld_addr http://$hostname:$vtctld_web_port/
+```
+
+### Additional parameters for running in prod
+
+```
+-queryserver-config-pool-size 24
```
[the remark this answers is not in the document]
Comment (sougou) [reply doc/VTTabletModes.md]: 24 for the read pool is usually good enough. It's the tx pool that should be in the 100s.

**Pull request merged**